DotNet.Bundle/PlistWriter.cs

```diff
@@ -86,6 +102,63 @@ private void WriteProperty(XmlWriter xmlWriter, string name, bool value)
             }
             
             xmlWriter.WriteEndElement();
+        }
+
+        private void WriteProperty(XmlWriter xmlWriter, string name, string[] values)
+        {
+            if (values.Length != 0)
+            {
+                xmlWriter.WriteStartElement("key");
+                xmlWriter.WriteString(name);
+                xmlWriter.WriteEndElement();
+
+                xmlWriter.WriteStartElement("array");
+                foreach (var value in values)
+                {
+                    if (!string.IsNullOrEmpty(value))
+                    {
+                        xmlWriter.WriteStartElement("string");
+                        xmlWriter.WriteString(value);
+                        xmlWriter.WriteEndElement();
+                    }
+                }
+
+                xmlWriter.WriteEndElement();
+            }
+        }
+
+
+        private void WriteProperty(XmlWriter xmlWriter, string name, ITaskItem[] values)
+        {
+            xmlWriter.WriteStartElement("key");
+            xmlWriter.WriteString(name);
+            xmlWriter.WriteEndElement();
+
+            xmlWriter.WriteStartElement("array");            
+
+            foreach (var value in values)
+            {
+                xmlWriter.WriteStartElement("dict");
+                var metadataDictionary = value.CloneCustomMetadata();
+
+                foreach (var entry in metadataDictionary.Keys)
```

Comment:
foreach (var entry in metadataDictionary)
{
    var key = entry.Key.ToString();
    var value = entry.Value.ToString();

    if (value.Contains(Separator.ToString()) || PropertiesWithMandatoryArray.Contains(entry))
    {
        WriteProperty(xmlWriter, entryString, value.Split(Separator));
    } else {
        WriteProperty(xmlWriter, entryString, value);
    }
}

Reply:
tried that first, but doesn't work.
![image](https://user-images.githubusercontent.com/22914881/116596390-30553100-a91c-11eb-9a3e-2469bc81790d.png)

Reply:
foreach (DictionaryEntry entry in metadataDictionary)